Projects/b_u585i_iot02a_ntz/Src/ota_pal/ota_pal_stm32u5_ntz.c

```diff
@@ -375,390 +591,565 @@ static HAL_StatusTypeDef prvWriteToFlash( uint32_t destination,
+                    case OTA_PAL_PENDING_ACTIVATION:
+                    case OTA_PAL_PENDING_SELF_TEST:
+                    case OTA_PAL_NEW_IMAGE_BOOTED:
+                    case OTA_PAL_NEW_IMAGE_WDT_RESET:
+
+                        if( ( prvSelectBank( ulGetOtherBank( pxContext->ulTargetBank ) ) == pdTRUE ) &&
+                            ( prvEraseBank( pxContext->ulTargetBank ) == pdTRUE ) &&
+                            ( prvDeleteImageNvContext() == pdTRUE ) )
+                        {
+                            xOtaStatus = OtaPalSuccess;
+                            pxContext->xPalState = OTA_PAL_READY;
+                        }
+
+                        break;
+
+                    default:
+                        break;
                 }
 
                 break;
 
             default:
-                LogError( ( "Unknown state transition, current state = %d, expected state = %d.\r\n", pContext->state, eState ) );
+                pxContext->xPalState = OTA_PAL_INVALID;
                 break;
         }
     }
 
-    return otaStatus;
+    return xOtaStatus;
 }
 
 
-OtaPalImageState_t otaPal_GetPlatformImageState( OtaFileContext_t * const pFileContext )
+OtaPalImageState_t otaPal_GetPlatformImageState( OtaFileContext_t * const pxFileContext )
 {
-    OtaPalImageState_t state = OtaPalImageStateUnknown;
-    OtaImageContext_t * pContext = prvGetImageContext();
+    OtaPalImageState_t xOtaState = OtaPalImageStateUnknown;
+    OtaPalContext_t * pxContext = prvGetImageContext();
 
-    switch( pContext->state )
+    ( void ) pxFileContext;
+
+    configASSERT( pxFileContext );
+
+    if( pxContext != NULL )
     {
-        case OTA_PAL_IMAGE_STATE_PENDING_COMMIT:
-            state = OtaPalImageStatePendingCommit;
-            break;
+        switch( pxContext->xPalState )
+        {
+            case OTA_PAL_NOT_INITIALIZED:
+            case OTA_PAL_INVALID:
+            default:
+                xOtaState = OtaPalImageStateUnknown;
+                break;
 
-        case OTA_PAL_IMAGE_STATE_COMMITTED:
-            state = OtaPalImageStateValid;
-            break;
+            case OTA_PAL_READY:
+            case OTA_PAL_FILE_OPEN:
+            case OTA_PAL_REJECTED:
+            case OTA_PAL_NEW_IMAGE_WDT_RESET:
+                xOtaState = OtaPalImageStateInvalid;
+                break;
 
-        case OTA_PAL_IMAGE_STATE_INVALID:
-            state = OtaPalImageStateInvalid;
-            break;
+            case OTA_PAL_PENDING_ACTIVATION:
+            case OTA_PAL_PENDING_SELF_TEST:
+            case OTA_PAL_NEW_IMAGE_BOOTED:
+                xOtaState = OtaPalImageStatePendingCommit;
+                break;
 
-        case OTA_PAL_IMAGE_STATE_PROGRAMMING:
-        case OTA_PAL_IMAGE_STATE_UNKNOWN:
-        default:
-            state = OtaPalImageStateUnknown;
-            break;
+            case OTA_PAL_ACCEPTED:
+                ( void ) prvSelectBank( pxContext->ulTargetBank );
```

Comment:
Why do we swap the bank when fetching the state of the image?

Reply:
Must have been a copy/paste mistake.